### PR TITLE
Remove NetworkCache and TokenNetworkContract

### PR DESCRIPTION
**Pull request by palango**
I ported the test network generation, ~~but it cannot find a path yet.~~

### CODE REVIEW

File: pathfinder/tests/fixtures/core.py [hunk 1]
```diff
@@ -1,73 +1,73 @@
...
 ) -> None:
 
-    def get_channel_deposits_patched(
-        self: TokenNetworkContract,
-        channel_id: ChannelId
-    )-> Dict[Address, int]:
+    for token_network in token_networks:
+        # some random magic to add channels
+        for channel_id in range(10):
+            random.seed(channel_id)
```
Comment (LefterisJP) [inline pathfinder/tests/fixtures/core.py]: You should not initialize the random seed more than once in a program. Just initialize it once somewhere in the program and either use the system time or leave it blank for usage os `os.urandom()`. Docs [here](https://docs.python.org/3.6/library/random.html#random.seed)

File: pathfinder/tests/fixtures/core.py [hunk 2]
```diff
@@ -1,73 +1,73 @@
...
-        get_channel_participants_patched
-    )
-    monkeypatch.setattr(
-        TokenNetworkContract,
-        'get_channel_deposits',
-        get_channel_deposits_patched
-    )
+        # deposit to channels
+        for channel_id in range(10):
+            random.seed(channel_id)
```
Comment (LefterisJP) [inline pathfinder/tests/fixtures/core.py]: same problem here

File: pathfinder/token_network.py [hunk 3]
```diff
@@ -38,40 +26,47 @@ class TokenNetwork:
...
     remain uncovered!
     """
 
     def __init__(
         self,
-        token_network_contract: TokenNetworkContract
+        token_network_contract: Contract
     ):
         """
         Initializes a new TokenNetwork.
```
Comment (LefterisJP) [inline pathfinder/token_network.py]: Keep this docstring in one line

File: pathfinder/token_network.py [hunk 4]
```diff
@@ -82,24 +77,49 @@ def handle_channel_new_deposit_event(
         """
         Register a new balance for the beneficiary.
 
-        Corresponds to the ChannelNewBalance event. Called by the contract event listener.
+        Corresponds to the ChannelNewDeposit event. Called by the contract event listener.
```
Comment (LefterisJP) [inline pathfinder/token_network.py]: No newline after docstring opening and before docstring closing

File: pathfinder/token_network.py [hunk 5]
```diff
@@ -82,24 +77,49 @@ def handle_channel_new_deposit_event(
...
+                )
+        except KeyError as ke:
+            log.error(
+                "Received ChannelNewDeposit event for unknown channel '{}'".format(
+                    channel_id
+                )
+            )
 
-    def handle_channel_closed_event(self, channel_id: ChannelId):
+    def handle_channel_closed_event(
```
Comment (LefterisJP) [inline pathfinder/token_network.py]: why did you change this? There is no reason to split it into multiple lines as long as it fits in one line. Fitting means < `99` columns.

File: pathfinder/tests/fixtures/core.py [hunk 6]
```diff
@@ -1,73 +1,73 @@
...
+    token_network_contracts: List[Contract],
 ) -> None:
 
-    def get_channel_deposits_patched(
-        self: TokenNetworkContract,
-        channel_id: ChannelId
-    )-> Dict[Address, int]:
+    for token_network in token_networks:
+        # some random magic to add channels
+        for channel_id in range(10):
```
Comment (devium) [inline pathfinder/tests/fixtures/core.py]: This is why it doesn't find any paths. You're only creating 10 channels over an address space of 100 different addresses (line 25). Set this to 100, same in line 37 (to add deposits to all channels), and reduce the number of nodes to something like 10 (line 25).

Apart from that I see Lefteris' issues with using `random.seed()` multiple times but this should be fixed in #17 once it's rebased on this PR. #17 pregenerates a set of private keys and addresses that are reused, so no need to generate them procedurally anymore.

So I'd say leave the `random.seed()` stuff in for now and we will fix it with #17.

**Comment by LefterisJP** [comment]
I am approving and mergin. Note: Made a tiny commit for style fixes.
